common/components/AddressField.tsx

```diff
@@ -14,23 +14,37 @@ interface Props {
 export const AddressField: React.SFC<Props> = ({ isReadOnly, isSelfAddress, isCheckSummed }) => (
   <AddressFieldFactory
     isSelfAddress={isSelfAddress}
-    withProps={({ currentTo, isValid, onChange, readOnly }) => (
-      <div className="input-group-wrapper">
-        <label className="input-group">
-          <div className="input-group-header">
-            {translate(isSelfAddress ? 'X_ADDRESS' : 'SEND_ADDR')}
-          </div>
-          <Input
-            className={`input-group-input ${isValid ? '' : 'invalid'}`}
-            type="text"
-            value={isCheckSummed ? toChecksumAddress(currentTo.raw) : currentTo.raw}
-            placeholder={donationAddressMap.ETH}
-            readOnly={!!(isReadOnly || readOnly)}
-            spellCheck={false}
-            onChange={onChange}
-          />
-        </label>
-      </div>
-    )}
+    withProps={({ currentLabel, currentTo, isValid, onChange, onFocus, onBlur, readOnly }) => {
+      let label;
+
+      if (currentLabel && !isSelfAddress) {
+        label = (
+          <span>
+            <i className="fa fa-check" /> {currentLabel}
+          </span>
+        );
```

Comment:
I'm not wild about changing the label for accessibility reasons. I'd prefer this be some green text below the input or something. Ideally this can be unified with the ENS resolver, and share the same styles. That one could use some UI love.

Reply:
:+1: